src/ui.c

```diff
@@ -1007,94 +1007,34 @@ static void DiskManagement(void)
-		UI_MENU_ACTION(CARTRIDGE_BLIZZARD_16, CARTRIDGE_BLIZZARD_16_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ATMAX_128, CARTRIDGE_ATMAX_128_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ATMAX_1024, CARTRIDGE_ATMAX_1024_DESC),
-		UI_MENU_ACTION(CARTRIDGE_SDX_128, CARTRIDGE_SDX_128_DESC),
-		UI_MENU_ACTION(CARTRIDGE_OSS_8, CARTRIDGE_OSS_8_DESC),
-		UI_MENU_ACTION(CARTRIDGE_OSS_043M_16, CARTRIDGE_OSS_043M_16_DESC),
-		UI_MENU_ACTION(CARTRIDGE_BLIZZARD_4, CARTRIDGE_BLIZZARD_4_DESC),
-		UI_MENU_ACTION(CARTRIDGE_AST_32, CARTRIDGE_AST_32_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ATRAX_SDX_64, CARTRIDGE_ATRAX_SDX_64_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ATRAX_SDX_128, CARTRIDGE_ATRAX_SDX_128_DESC),
-		UI_MENU_ACTION(CARTRIDGE_TURBOSOFT_64, CARTRIDGE_TURBOSOFT_64_DESC),
-		UI_MENU_ACTION(CARTRIDGE_TURBOSOFT_128, CARTRIDGE_TURBOSOFT_128_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ULTRACART_32, CARTRIDGE_ULTRACART_32_DESC),
-		UI_MENU_ACTION(CARTRIDGE_LOW_BANK_8, CARTRIDGE_LOW_BANK_8_DESC),
-		UI_MENU_ACTION(CARTRIDGE_SIC_128, CARTRIDGE_SIC_128_DESC),
-		UI_MENU_ACTION(CARTRIDGE_SIC_256, CARTRIDGE_SIC_256_DESC),
-		UI_MENU_ACTION(CARTRIDGE_SIC_512, CARTRIDGE_SIC_512_DESC),
-		UI_MENU_ACTION(CARTRIDGE_STD_2, CARTRIDGE_STD_2_DESC),
-		UI_MENU_ACTION(CARTRIDGE_STD_4, CARTRIDGE_STD_4_DESC),
-		UI_MENU_ACTION(CARTRIDGE_RIGHT_4, CARTRIDGE_RIGHT_4_DESC),
-		UI_MENU_ACTION(CARTRIDGE_BLIZZARD_32, CARTRIDGE_BLIZZARD_32_DESC),
-		UI_MENU_ACTION(CARTRIDGE_MEGAMAX_2048, CARTRIDGE_MEGAMAX_2048_DESC),
-		UI_MENU_ACTION(CARTRIDGE_THECART_128M, CARTRIDGE_THECART_128M_DESC),
-		UI_MENU_ACTION(CARTRIDGE_MEGA_4096, CARTRIDGE_MEGA_4096_DESC),
-		UI_MENU_ACTION(CARTRIDGE_MEGA_2048, CARTRIDGE_MEGA_2048_DESC),
-		UI_MENU_ACTION(CARTRIDGE_THECART_32M, CARTRIDGE_THECART_32M_DESC),
-		UI_MENU_ACTION(CARTRIDGE_THECART_64M, CARTRIDGE_THECART_64M_DESC),
-		UI_MENU_ACTION(CARTRIDGE_XEGS_8F_64, CARTRIDGE_XEGS_8F_64_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ATRAX_128, CARTRIDGE_ATRAX_128_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ADAWLIAH_32, CARTRIDGE_ADAWLIAH_32_DESC),
-		UI_MENU_ACTION(CARTRIDGE_ADAWLIAH_64, CARTRIDGE_ADAWLIAH_64_DESC),
-		UI_MENU_END
-	};
-
-	int i;
+	/* menu_array must be terminated with UI_MENU_END */
+	UI_tMenuItem menu_array[CARTRIDGE_LAST_SUPPORTED+1]; 
+	int cart_entry, menu_entry;
 	int option = 0;
+	UWORD flag;
 
 	UI_driver->fInit();
 
-	for (i = 1; i <= CARTRIDGE_LAST_SUPPORTED; i++)
-		if (CARTRIDGE_kb[i] == k) {
+	for (cart_entry = 1; cart_entry <= CARTRIDGE_LAST_SUPPORTED;
+	     cart_entry++) {
+		/* N.B.: CARTRIDGES[] is offset by 1 from menu_array[]
+		   due to CARTRIDGE_NONE entry. */
+		menu_entry = cart_entry - 1;
+
+		if (CARTRIDGES[cart_entry].kb == k) {
 			if (option == 0)
-				option = i;
-			menu_array[i - 1].flags = UI_ITEM_ACTION;
+				option = cart_entry;
+			flag = UI_ITEM_ACTION;
 		}
 		else
-			menu_array[i - 1].flags = UI_ITEM_HIDDEN;
+			flag = UI_ITEM_HIDDEN;
 
+		UI_MAKE_MENU_ACTION(menu_array[menu_entry], flag, cart_entry,
+				    CARTRIDGES[cart_entry].description);
```

Comment:
This file is formatted with tabs of width 4 (see the last line in the file), so this does not look aligned.

Reply:
Sorry, are we talking about tab-stops in my deltas?  Or are you looking at the way the line break in UI_MAKE_MENU_ACTION() is aligned?

Reply:
Yes, in my editor, the "CARTRIDGES" line is not aligned with "menu_" above, as it uses tab width = 4, because at the end of the file it says "vim:ts=4:sw=4:"

Sadly, the Atari800 sources use a mixture of tab and spaces of different widths all over the place.

Reply:
OK, that's just how my emacs is setup. It's an easy fix. :)